src/provider/camunda-platform/properties/ProcessVariablesProps.js

```diff
@@ -24,31 +24,36 @@ import {
   useService
 } from '../../../hooks';
 
+import { ListGroup } from '@bpmn-io/properties-panel';
+import { useEffect, useState } from '@bpmn-io/properties-panel/preact/hooks';
 
-export function ProcessVariablesProps(props) {
 
+function ProcessVariablesEntry(props) {
   const {
     element
   } = props;
 
-  if (!canHaveProcessVariables(element)) {
-    return null;
-  }
 
-  const businessObject = getBusinessObject(element);
-  const rootElement = getRootElement(businessObject);
-  const scope = getScope(element);
+  const [ variables, setVariables ] = useState([]);
+
+  useEffect(async () => {
+    const businessObject = getBusinessObject(element);
+    const rootElement = getRootElement(businessObject);
+    const scope = getScope(element);
+
+    const rawVariables = await getVariablesForScope(scope, rootElement);
+
+    const withName = populateElementNames(sortByName(rawVariables));
+
+    setVariables(withName);
+  }, [ element ]);
 
-  // (1) fetch available process variables for given scope
-  const variables = getVariablesForScope(scope, rootElement);
 
   if (!variables.length) {
     return null;
```

Comment:
Does this still work in the async usage case?

Reply:
I think I understand now, but just to confirm: 

* We do a shallow test to understand if an element may have process variables. If so, then we return a component to render the variables.
* That component renders something _only_ when `variables` exist

Does it make sense to render an empty state instead, in the unlikely case that no process variables exists?

Reply:
> Does it make sense to render an empty state instead, in the unlikely case that no process variables exists?

We do not have an concept for empty states in Groups, so they always need content. What we could do is use a normal Group and add `No variables available in this scope` as content.

However, this makes it hard to see that no Variables are present from the collapsed state, as we never show a count of 0.  I decided against it and kept the current behavior for now, where the group is not shown at all when no variables are present.

Reply:
Thanks for elaborating. Makes sense to me.